src/G2/Execution/Rules.hs

```diff
@@ -235,7 +235,10 @@ retLam s@(State { expr_env = eenv })
 
 traceType :: E.ExprEnv -> Expr -> Maybe Type
 traceType _ (Type t) = Just t
-traceType eenv (Var (Id n _)) = traceType eenv =<< E.lookup n eenv
+traceType eenv (Var (Id n _)) = case E.lookupConcOrSym n eenv of 
+                                        Just (E.Sym i) -> Just (TyVar i)
```

Comment:
Might be good to have a comment explaining what is happening here?  I think if we are trying to look up a type, and get a symbolic variable, then it must be a symbolic type variable?

Reply:
"varabile" -> "variable", "symoblic" -> "symbolic"